### DISCUSSION TITLE
Map not found for new location

### DISCUSSION THREAD

**Discussion by Spsrky** [code4recovery/12-step-meeting-list#864] [Q&A]
Problem with entering a new meeting at a new location: After entering meeting title, date/time, etc. adding a name and address for a new location is not recognized - no map appears and the Publish button remains grayed out. If a current existing location is used, the meeting can be saved. The new address was confirmed as a valid location by checking with Google Maps. We are currently using Mapbox and have the Mapbox key plugged in on the TSML Settings page. I've tried several new addresses with the same problem. Reinstalling the plugin didn't correct this.

**Comment by brianw-area24** [comment]
I'm thinking this is the same issue as #863, try removing the ", USA" from the end of the address, then going to another field in the form, like region. This should re-geocode the address and cause the publish button to work again.

**Reply by Spsrky** [reply]
Thanks, Brian - no luck with that; I'e tried changing the location address several different ways. 

A bit of additional information: I downloaded the CSV file of meetings, then uninstalled and reinstalled the TSML plugin and uploaded meetings.csv. Our total number of meetings in the last went from about 1100 to about 800 and the upload created about 300 error messages like this one:

```
Google gave an unexpected response for address 9731 SE King Rd, Portland, OR 97222, USA. Response was (object) array(
	30	   'error_message' => 'You must enable Billing on the Google Cloud Project at https://console.cloud.google.com/project/_/billing/enable Learn more at https://developers.google.com/maps/gmp-get-started',
	31	   'results' => 
	32	  array (
	33	  ),
	34	   'status' => 'REQUEST_DENIED',

```
This was puzzling, since we've been using a MapBox API, not Google Maps. Now I'm going the (rather involved) process of getting Google Maps hooked up; using our Google Non-profits account, there is no charge for the Maps service.

We've been having map and location problems for a couple of months and it now seems that we may have exceeded some limit at around 800 records. Once we get the Google Maps account finally active, I'll post the results.

**Reply by tim-rohrer** [reply]
@Spsrky Google Maps does not equal Google Maps Geocoding. I know it is confusing, but even when you have MapBox enabled, your TSML is still going to use our API key for conducting geocoding.

You're the second or third case where we've seen this recently. Something is going on, but we don't know where yet. We did (for security reasons) "expire" older Google keys, but this was quite some time ago.

What version of TSML are you using?

**Reply by Spsrky** [reply]
v. 3.14.5

**Reply by tim-rohrer** [reply]
I resolved the issue with Code for Recovery's Google account earlier this evening. This should have resolved the geocoding issue that ws leading to a greyed out publish button and no visible map when trying to add a new location.

Please try the process again. It doesn't matter if you have a valid Google or Mapbox key/token for mapping, as long as you have one of them. 

The problem was with geocoding (and lack of a visible map was a second order effect). If you still get the greyed out publish button, please open a browser debugging console and try entering the address again. Watch for "Geocoded" to show up in the console. Then expand that, screen cap it, and post it here.

Thanks.

**Comment by lionssharedigital** [comment]
Nashville is having this same problem. He is unable to add a new location or address as described. I'm happy to work with you to get his resolved or access to their systems. 
Using latest versions of WordPress and latest TSML

**Reply by tim-rohrer** [reply]
Feel free to try using the beta api geocoding gateway as an interim.

Meetings-->Import & Settings-->Settings:

<img width="339" alt="image" src="https://user-images.githubusercontent.com/7756865/178841042-d6e350fd-993a-4f94-9e42-b3a6321c0f73.png">

**Reply by teach4learning** [reply]
This solved our problem in Corpus Christi, TX

**Reply by Spsrky** [reply]
The beta appears to have solved the problem for us in Portland, too - thanks!

**Reply by tim-rohrer** [reply]
Legacy should work as well. If not, please let me know.

**Comment by teach4learning** [comment]
If I switched to the Beta, should I switch back to Legacy?

**Reply by tim-rohrer** [reply]
You can use either.

**Comment by Spsrky** [comment]
Hi Tim,

You're right, that is confusing. We've applied for Google Non-Profit support for an API key with expanded usage, hoping that will remove the apparent restriction. Maybe?

We are currently using v. 3.14.5, we try to stay current.

Thanks,

Theron

**Reply by tim-rohrer** [reply]
Are you experiencing a new issue? 

We strive to make it so users do not have to apply for a any Google key unless they wish to use Google maps vice MapBox maps. Users do need to register for (and activate) either a MapBox or a Google key for *mapping*. However, we provide geocoding, and users shouldn’t have to create any account with Google for this service.

What is the restriction you’re facing? If this is indeed the same issue as your original, then by all means we should continue in this thread. But if it is new, then I suggest starting a new discussion.

In either case, please include screen caps of some of your settings related to geocoding to help us better understand your issues.
